src/rgw/rgw_user.cc

```diff
@@ -1391,13 +1391,18 @@ int RGWSubUserPool::add(RGWUserAdminOpState& op_state, std::string *err_msg, boo
 {
   std::string subprocess_msg;
   int ret;
+  int32_t key_type = op_state.get_key_type();
 
   ret = check_op(op_state, &subprocess_msg);
   if (ret < 0) {
     set_err_msg(err_msg, "unable to parse request, " + subprocess_msg);
     return ret;
   }
 
+  if (key_type == KEY_TYPE_S3 && op_state.get_access_key().empty()) {
```

Comment:
@snakeAngel2015: Isn't there a way to specify 'gen_access-key' in the command? something like - ./s3curl.pl --id=personal --put – http://radosgw.h3c.com:8000/admin/user?subuser\&uid=yrf2\&subuser=yrf2:yrf1\&key-type=s3\&gen-access-key=true -v.
Looks like the code expects the access key or the gen-access-key for the access-key : https://github.com/ceph/ceph/blob/master/src/rgw/rgw_user.cc#L843. The radosgw-admin cli complains if neither is specified instead of generating an access key by default